pkg/server/http/ipfs.go

```diff
@@ -186,30 +188,41 @@ func ipfsHandler(lassie *lassie.Lassie, cfg HttpServerConfig) func(http.Response
 			request.MaxBlocks = blockLimit
 		}
 
+		// servertiming metrics
+		startTimeMap := make(map[string]time.Time)
+
 		logger.Debugw("fetching CID", "retrievalId", retrievalId, "CID", rootCid.String(), "path", path.String(), "dagScope", dagScope)
 		stats, err := lassie.Fetch(req.Context(), request, func(re types.RetrievalEvent) {
 			header := servertiming.FromContext(req.Context())
 			if header == nil {
 				return
 			}
 
+			var metricName string
+			switch re.(type) {
+			case events.StartedFindingCandidatesEvent:
+				header.NewMetric("indexer")
+				startTimeMap["indexer"] = re.Time()
+
+			case events.StartedRetrievalEvent:
+				header.NewMetric("retrieval")
+				startTimeMap["retrieval"] = re.Time()
+				metricName = "indexer"
+
+			case events.FinishedEvent:
+				metricName = "retrieval"
+			}
+
+			// Set the metric duration
 			header.Lock()
 			if header.Metrics != nil {
 				for _, m := range header.Metrics {
-					if m.Name == string(re.Phase()) {
```

Comment:
This code appears to be missing from the changes. @willscott is this ok -- I think you put it in here is anyone using it?

Reply:
I did reference the behavior shown in this PR while making the changes to the server timings.

Because the events were touched, the entire ServerTimings header will be slightly different. I kept the "indexer" and "retrieve" metrics. The "fetch" metric duration would have always been close to 0 and unhelpful due to the way this was calculating the fetch phase duration by taking a diff between it and the indexer phase. The fetch phase was a parent phase, not a serial phase, so it was inaccurate to begin with. The extra events in each phase might change since we're using different events.

Reply:
I have put together PR #332 to address the server timing header. After talking to @willscott some more, the new logic was going to be a bit more cumbersome. Some more thoughts and an example of the new `Server-Timing` value can be found over there.